### ISSUE TITLE
Teams animations setting - spike how to support in motion components

### ISSUE THREAD

**Issue by robertpenner** [microsoft/fluentui#33113]
**Status**
This task is currently waiting on the completion of two dependent tasks:

 [ ] Implementation of a mechanism to **disable motion per component instance** (e.g., for drag-and-drop Tree feature).
 [ ] Guidance for developers on how to disable motion where needed

Reference: https://github.com/orgs/microsoft/projects/1222/views/3?pane=issue&itemId=82092928

-----------------------------

Tasks:
- [ ] Validation with Fluent UI developers

## Problem Background
 
- Our motion components read the preference for reduced motion from the browser, which is automatically in sync with the OS setting (e.g. "**Animation effects**" in Windows):

![Image](https://github.com/user-attachments/assets/74b30329-47de-46fb-ac95-9097413ae211)

- However, Teams has its own setting to "**Turn off animations**", and this Teams setting is not connected to the OS setting:

![Image](https://github.com/user-attachments/assets/baf76434-14e2-4aaf-95d7-7083f07c8175)

- Thus, our motion components currently do not respond to the Teams settings to turn off animations.

## Solution to Investigate

- Teams has a `useDisableAnimations` [hook](https://domoreexp.visualstudio.com/Teamspace/_git/teams-modular-packages?path=%2Fpackages%2Fcomponents%2Fcomponents-core-theme-provider%2Fsrc%2Fuse-disable-animations.tsx&version=GBmaster) for consuming the Teams motion toggle.
- However, the motion components should not consume this Teams hook directly:
  - Architecturally, our components are in Fluent, so they need to be app-agnostic, not reach out to look for specific app settings.
  - Instead, Teams code should read the setting and configure the motion components from the outside-in.
- Lingfan [added](https://github.com/microsoft/fluentui/pull/32316) a new mechanism `MotionBehaviourContext` which can tell all motion components within it to disable motions, via React context. 
- Lingfan's immediate use case (for Simple Collab Left Rail) was to temporarily disable motion in Tree, during drag and drop. However, we would like to use the same mechanism at the app level to disable motion for any motion component within Teams.
- Prior art: there is a place in Teams code where the Animation component from Fluent v0 (Stardust) is configured to disable animations from the outside in, using a context provider:

![Image](https://github.com/user-attachments/assets/8be56701-abab-430a-95fe-f3289070f2c6)

- We can follow this precedent to try a spike, to feed the Teams setting for disabling motions into a new `MotionBehaviourContext`, wrapping around Teams at its base.

## Testing Considerations

- We should confirm with Teams Engineering the intended behavior: whether the Teams motion toggle should completely ignore the OS setting, and override it.
- `MotionBehaviourContext` inside a `MotionBehaviourContext`: we should check that overriding works as desired, e.g. that the app-level setting and the drag-and-drop use case (see above) work in harmony. 

## Future Work

- Shift and Lingfan would like to see this Teams motion toggle removed, in favor of relying solely on the OS setting.
- Shift mentioned the Teams PM for Accessibility is Brett Humphrey, who would be involved in a decision to remove the Teams motion toggle to rely on the OS setting.
